src/screens/Platform/DAppDisclaimer.js

```diff
@@ -0,0 +1,223 @@
+// @flow
+
+import React, { useCallback, useState } from "react";
+
+import { useTheme } from "@react-navigation/native";
+import { Trans, useTranslation } from "react-i18next";
+import { StyleSheet, View, TouchableOpacity } from "react-native";
+
+import BottomModal from "../../components/BottomModal";
+import Checkbox from "../../components/CheckBox";
+import Button from "../../components/Button";
+import LiveIcon from "../../icons/LiveLogo";
+import LText from "../../components/LText";
+import CloseIcon from "../../icons/Close";
+import InfoIcon from "../../icons/Info";
+import { rgba } from "../../colors";
+import AppIcon from "./AppIcon";
+
+export type Props = {
+  closeDisclaimer: () => void,
+  disableDisclaimer: () => void,
+  onContinue: () => void,
+  isOpened: boolean,
+  icon?: string | null,
+};
+
+const DAppDisclaimer = ({
+  closeDisclaimer,
+  isOpened,
+  disableDisclaimer,
+  onContinue: next,
+  icon,
+}: Props) => {
+  const { t } = useTranslation();
+  const { colors } = useTheme();
+  const [
+    disableDisclaimerChecked,
+    setDisableDisclaimerChecked,
+  ] = useState<boolean>(false);
+
+  const onClose = useCallback(() => {
+    closeDisclaimer();
+  }, [closeDisclaimer]);
+  const onContinue = useCallback(() => {
+    if (disableDisclaimerChecked) {
+      disableDisclaimer();
+    }
+    closeDisclaimer();
+    next();
+  }, [disableDisclaimerChecked, closeDisclaimer, disableDisclaimer, next]);
+
+  return (
+    <BottomModal
+      style={{
+        ...styles.root,
+        backgroundColor: colors.card,
+      }}
+      isOpened={isOpened}
+    >
+      <TouchableOpacity style={styles.closeIcon} onPress={onClose}>
+        <CloseIcon size={18} />
+      </TouchableOpacity>
+
+      <View style={[styles.flowRow, styles.head]}>
+        <View style={[styles.headElements, styles.roundLogoContainer]}>
+          <LiveIcon size={31} color="blue" />
+        </View>
+        {icon && (
+          <>
+            <View style={[styles.headElements, styles.flowRow]}>
+              {Array(6)
+                .fill()
+                .map((_, i) => (
+                  <View
+                    key={i}
+                    style={{ ...styles.dash, backgroundColor: colors.live }}
+                  />
+                ))}
+            </View>
```

Comment:
You should memoize this "dashs" component to avoid this array.map on each render.

```suggestion
            <View style={[styles.headElements, styles.flowRow]}>{Dashs}</View>
```

Here is one way to do so:
```js
  const Dashs = useMemo(
    () =>
      [...Array(6)].map((_, i) => (
        <View
          key={i}
          style={{ ...styles.dash, backgroundColor: colors.live }}
        />
      )),
    [colors.live],
  );
```

Reply:
IMO the cost of memoization is way worse than 2 consecutive loops on 6 indexes. Memoization should be done either on state related data in order to work with the hooks api or for performance consuming tasks, and to me this is none of those case.

If we really want to avoid these loops, I can always declare the array outside of the component, but it feels pretty nitpicky to me 😜

Reply:
It looks like it could still be optimised because as of today, this whole logic (computation + render) is done every time an action happens.

cf. 
<details>
<summary>Before memoization</summary>

https://user-images.githubusercontent.com/87011321/126509064-d9ba7579-ee00-484a-a1a7-fb78aa86ac66.mov

</details>

<details>
<summary>After memoization</summary>

https://user-images.githubusercontent.com/87011321/126509043-131ed775-0f82-4973-ac69-fb356fc04ae9.mov

</details>

It would be interesting to run the Profiler to test any performance drop or optimisation related to this.

Reply:
Yeah that's not really what I meant by the cost of memoization. I was talking about this kind of point: https://kentcdodds.com/blog/usememo-and-usecallback.
Sure we are re-rendering 6 empty "divs" pretty much, but the cost of "optimizing" this with a memoization will probably (surely) have no benefit on performances and might actually make it worse from what I can tell.

I will memoize it to make your logs disappear but I believe it won't have the effect you think it has 😆